src/chunk.c

```diff
@@ -930,8 +932,8 @@ ts_chunk_get_data_node_name_list(const Chunk *chunk)
  * the chunk.
  */
 static Chunk *
-chunk_create_metadata_after_lock(Hypertable *ht, Hypercube *cube, const char *schema_name,
-								 const char *table_name, const char *prefix)
+chunk_create_object(const Hypertable *const ht, Hypercube *const cube, const char *schema_name,
```

Comment:
Out of curiosity, why do you bother with `const` pointers?

I do get why it makes sense to have `const` objects in the parameter list since it signals no side-effects on these objects. But `const` pointers seem almost more trouble than it is worth, especially the way it is applied here.

A couple of observations and thoughts on this:

* Since the pointer is passed by value it has no side effects to set it in the function. 
* Often the use of `const` pointers is inconsistent. For instance, why isn't at least `table_name` also a `const` pointer when you applied it to most of the other parameters (it doesn't seem to be set in the function). Another example: In the parameter list of `chunk_insert_into_metadata_after_lock` you made the object `const` but not the pointer.
* Why does this pattern seem to apply to only function parameters but not, e.g., function variable pointers? If it is really important to not change pointers, it should apply also to pointer variables and pointers to non-const objects. Am I missing something?
* Personally, I don't find a whole lot of value in making pointers `const`; in fact, it mostly makes the code much more verbose. I do think we should be better at making objects `const` though.
* I can't remember having seen any C projects/code that use `const` pointers this way, probably for the reasons I listed.

Interested to hear your thoughts on this. We can discuss separately, but just wanted to leave my thoughts on this while in context.

Reply:
The second const means that the variable cannot be reused for another pointer of the same type.

Reply:
I agree with erik here this just adds line noise and additional development overhead with little practical benefit

Reply:
It is not a noise. In general, it provides semantics how the variable will be used in the code. I.e., that it cannot be reused to store different value. Thus I personally prefer to see `const` more often, so it is clear when to expect variable reuse.
There is a drawback that it doesn't play well with old Microsoft compilers.

Reply:
> * why isn't at least `table_name` also a `const` pointer when you applied it to most of the other parameters (it doesn't seem to be set in the function).

@erimatnor `table_name` is set in the function when it is not provided. So this actually demonstrates that it is good to be explicit about using `const` as you missed to find the usage in the code.

Reply:
Since this function is long, I find it useful to know about variables reuse. I haven't changed other functions as they are simple enough.